server/utils/FirebaseAlgoliaServer/functions/tests/integrationTestHelpers.js

```diff
@@ -0,0 +1,311 @@
+ */
+async function checkEntityEquals(entityId, expectedData) {
+  const entity = await db.collection(COLLECTION).doc(entityId).get();
+
+  if (!entity.exists) {
+    throw(`Document ${entityId} firestore retrieval FAILURE`);
+  } else {
+    const entityData = entity.data();
+    const loggingString = 
+        `\nExpected ${JSON.stringify(expectedData)}\nGot ${JSON.stringify(entityData)}\n`;
+    if (!isEquivalent(expectedData, entityData)){
+      throw(loggingString); 
+    }
+    console.log(loggingString);
+    console.log("SUCCESS.\n");
+  }
+}
+
+/**
+ * Wrapper for checkEntityEquals
+ * If the entity doesn't match expected data, catches the error
+ * @param {!Promise=} entityPromise promise returned when entity was made
+ * @param {json=} expectedData expected DB entry
+ */
+function checkEntityEqualsFromPromise(entityPromise, expectedData) {
+  entityPromise.then(promiseValue => {
+    checkEntityEquals(promiseValue.id, expectedData)
+    .catch((error) => console.log("Equality check FAILURE", error));
+  });
+}
+
+// Algolia helper functions 
+
+/**
+ * Retrieve an entity from Algolia as a JSON object if it exists
+ * @param {number=} objectID primary key, equal to entityId
+ * @return {!Promise}
+ */
+function getObjectFromAlgolia(objectID) {
+  return index.getObject(objectID)
+              .catch((error) => {
+                console.log(`Couldn't retrieve ${objectID} because ${error}`)
+              });
+}
+
+/**
+ * Wrapper for getObjectFromAlgolia
+ * if the object is empty/null/doesn't exist etc, catches as an error
+ * @param {!Promise=} promise promise returned when entity was made in firebase
+ * @return {!Promise}
+ */
+function getObjectFromAlgoliaWithPromise(promise) {
+  const algoliaPromise = promise.then(promiseValue => getObjectFromAlgolia(promiseValue.id));
+  algoliaPromise.then(object => {
+    if (typeof object !== 'undefined') {
+      console.log(`Found ${JSON.stringify(object.data)}`);
+    } else {
+      throw("Malformed object returned from algolia");
+    }
+  }).catch((error) => console.log(error));
+  
+  return algoliaPromise; 
+}
+
+/**
+ * Checks that the algolia record was deleted 
+ * Assuming original entity deleted in firestore
+ * @param {!Promise=} promise promise returned when entity was made in firebase
+ * @return {!Promise}
+ */
+async function checkObjectDeletedFromAlgoliaWithPromise(promise) {
+  await sleep(DEFAULT_SLEEP*2); // Wait for deletion to propogate
+  const algoliaPromise = 
+      getObjectFromAlgoliaWithPromise(promise).then(object => {
+        throw("Algolia object should not exist but it does")
+      }).catch(function onError(error) {
+        console.log("Above malformed message normal. Algolia delete SUCCESS.")
+        // 'Above message' is an unavoidable 'Malformed object ...' error
+        // The only way to test for existence is to call for it directly
+        // And when it isn't found, an error has to be thrown... then caught
```

Comment:
Try writing a COUNT statement to count the number of entities that match some condition, and verify that it is 0.

What you have here is error-prone because if the query fails for a different reason the test will still pass.